include/boost/fusion/support/detail/and.hpp

```diff
@@ -20,7 +20,11 @@ namespace boost { namespace fusion { namespace detail {
 
     template<typename ...T>
     struct and_impl<integral_constant<T, true>...> : true_type {};
-   
+
+    // This specialization is necessary to avoid MSVC-12 variadics bug.
```

Comment:
Variadics vector is [disabled on MSVC 12](https://github.com/boostorg/fusion/blob/develop/include/boost/fusion/container/vector/detail/config.hpp#L31), should this be reverted and the test disabled for this compiler instead?